frontend-react/src/TelemetryService.test.ts

```diff
@@ -0,0 +1,325 @@
+import {
+    ApplicationInsights,
+    SeverityLevel,
+} from "@microsoft/applicationinsights-web";
+
+jest.mock("@microsoft/applicationinsights-web", () => {
+    return {
+        ...jest.requireActual("@microsoft/applicationinsights-web"),
+        ApplicationInsights: function () {
+            return {
+                loadAppInsights() {},
+                trackEvent: jest.fn(),
+                trackException: jest.fn(),
+                addTelemetryInitializer: jest.fn(),
+                context: {
+                    getSessionId() {
+                        return "test-session-id";
+                    },
+                },
+            };
+        },
+    };
+});
+
+describe("TelemetryService", () => {
+    const oldConnectionString =
+        process.env.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING;
+
+    let ai: { initialize: () => void };
+    let getAppInsights: () => ApplicationInsights;
+    let getAppInsightsHeaders: () => object;
+    let withInsights: (console: Console) => void;
+
+    beforeEach(() => {
+        // Isolating modules to "reset" previously initialized appInsights, reactPlugin, etc.
+        jest.isolateModules(() => {
+            ({
+                ai,
+                getAppInsights,
+                getAppInsightsHeaders,
+                withInsights,
+            } = require("./TelemetryService"));
+        });
+    });
+
+    describe("#initialize", () => {
```

Comment:
Backfilled a bunch of tests along the way!  The ones in [this describe](https://github.com/CDCgov/prime-reportstream/pull/7779/files#diff-d9d00c1fd91c8fd306ae3d32c0ecd6620d304cf3638678f8bfacf3ff9f5285beR136) are the only ones that are actually relevant to this changeset, but feel free to review as needed!